keps/sig-cluster-lifecycle/kubeadm/20190916-kubeadm-operator.md

```diff
@@ -0,0 +1,363 @@
+---
+title: Kubeadm operator
+authors:
+  - "@fabriziopandini"
+owning-sig: sig-cluster-lifecycle
+participating-sigs:
+  - sig-cluster-lifecycle=
+reviewers:
+  - "@neolit123"
+  - "@rosti"
+  - "@ereslibre"
+  - "@detiber"
+  - "@vincepri"
+  - "@yastij"
+  - "@chuckha"
+approvers:
+  - "@timothysc"
+  - "@luxas"
+editor: "@fabriziopandini"
+creation-date: 2019-09-16
+last-updated: 2019-09-16
+status: implementable
+---
+
+# Kubeadm operator
+
+## Table of Contents
+
+<!-- toc -->
+- [Release Signoff Checklist](#release-signoff-checklist)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+    - [Story 3](#story-3)
+    - [Story 4](#story-4)
+    - [Story 5](#story-5)
+  - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+    - [Action](#action)
+    - [Action Controller and TaskGroup](#action-controller-and-taskgroup)
+    - [TaskGroupController and Task](#taskgroupcontroller-and-task)
+    - [TaskController](#taskcontroller)
+    - [Observability and Operability](#observability-and-operability)
+      - [Execution order](#execution-order)
+      - [Execution modes](#execution-modes)
+      - [Error management](#error-management)
+    - [Extensibility](#extensibility)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed](#infrastructure-needed)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [x] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [x] KEP approvers have set the KEP status to `implementable`
+- [x] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+## Summary
+
+Kubeadm operator would like to enable declarative control of kubeadm workflows, automating the execution and the orchestration of such tasks across existing nodes in a cluster.
```

Comment:
can the list of tasks be specified or some examples be given in the summary?

Reply:
The KEP specification recommend short Summary; instead in the motivation I'm giving more detail and linking the kubeadm upgrade page as a example